File: packages/EvoHtmlToPdf-v4.12/Samples/NET_2.0/VB/WinForms/EvoHtmlToPdfDemo/HtmlInHeaderAndFooter.vb
Imports System.ComponentModel
Imports System.Text
Imports System.IO

Imports EvoPdf


Namespace EvoHtmlToPdfDemo
	Partial Public Class HtmlInHeaderAndFooter
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnConvert_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnConvertMerge.Click
			Me.Cursor = Cursors.WaitCursor

			Try
				Dim pdfConverter As New PdfConverter()

				' set the license key
                pdfConverter.LicenseKey = "B4mYiJubiJiInIaYiJuZhpmahpGRkZE="

				' show header and footer in the rendered PDF
				pdfConverter.PdfDocumentOptions.ShowHeader = True
				pdfConverter.PdfDocumentOptions.ShowFooter = True

				' set the header height in points
				pdfConverter.PdfHeaderOptions.HeaderHeight = 60

                ' set the header HTML area
                Dim headerHtml As HtmlToPdfElement = New HtmlToPdfElement(0, 0, 0, pdfConverter.PdfHeaderOptions.HeaderHeight,
                    Path.Combine(Application.StartupPath, "..\..\HeaderFooter\HeaderAndFooterHtml.htm"), 1024, 0)
                headerHtml.FitHeight = True
                pdfConverter.PdfHeaderOptions.AddElement(headerHtml)

                ' set the footer height in points
                pdfConverter.PdfFooterOptions.FooterHeight = 60
                'write the page number
                Dim footerTextElement As TextElement = New TextElement(0, 30, "This is page &p; of &P;  ",
                    New Font(New FontFamily("Times New Roman"), 10, GraphicsUnit.Point))
                footerTextElement.TextAlign = HorizontalTextAlign.Right
                pdfConverter.PdfFooterOptions.AddElement(footerTextElement)

                ' set the footer HTML area
                Dim footerHtml As HtmlToPdfElement = New HtmlToPdfElement(0, 0, 0, pdfConverter.PdfFooterOptions.FooterHeight,
                    Path.Combine(Application.StartupPath, "..\..\HeaderFooter\HeaderAndFooterHtml.htm"), 1024, 0)
                footerHtml.FitHeight = True
                pdfConverter.PdfFooterOptions.AddElement(footerHtml)

				' save the PDF bytes in a file on disk
				Dim outFilePath As String = Path.Combine(Application.StartupPath, "HtmlInHeaderAndFooter.pdf")

				If Not cbAlternateHeaderAndFooter.Checked Then
					' the header content is the same on all the PDF pages
					' the footer content is the same on all the PDF pages
					pdfConverter.SavePdfFromUrlToFile(textBoxURL1.Text, outFilePath)
				Else
					' set an alternate header and footer on the even pages

					' call the converter and get a Document object from URL
					Dim pdfDocument As Document = pdfConverter.GetPdfDocumentObjectFromUrl(textBoxURL1.Text)

					If pdfDocument.Pages.Count >= 2 Then
						' get the alternate header and footer width and height
						' the width is given by the PDF page width
						Dim altHeaderFooterWidth As Single = pdfDocument.Pages(0).ClientRectangle.Width
						' the height is the same with the document header height from the PdfConverter object
						Dim altHeaderHeight As Single = pdfConverter.PdfHeaderOptions.HeaderHeight
						Dim altFooterHeight As Single = pdfConverter.PdfFooterOptions.FooterHeight

						' create the alternate header template
						Dim altHeaderTemplate As Template = pdfDocument.Templates.AddNewTemplate(altHeaderFooterWidth, altHeaderHeight)

						' add html to the header
						Dim altHeaderHtml As New HtmlToPdfElement(0, 0, 0, pdfConverter.PdfHeaderOptions.HeaderHeight, Path.Combine(Application.StartupPath, "..\..\HeaderFooter\HeaderAndFooterHtml2.htm"), 1024, 0)
						altHeaderHtml.FitHeight = True
						altHeaderTemplate.AddElement(altHeaderHtml)

						' add a horizontal line to the bottom of the header
						Dim headerLine As New LineElement(0, altHeaderHeight, altHeaderFooterWidth, altHeaderHeight)
						altHeaderTemplate.AddElement(headerLine)

						' add page numbering to the left of the header
						Dim pageNumberFont As PdfFont = pdfDocument.Fonts.Add(New Font(New FontFamily("Times New Roman"), 10, GraphicsUnit.Point))
						Dim pageNumbering As New TextElement(10, 10, "Page &p; of &P;", pageNumberFont, Color.Blue)

						altHeaderTemplate.AddElement(pageNumbering)

						' create the alternate footer template
						Dim altFooterTemplate As Template = pdfDocument.Templates.AddNewTemplate(altHeaderFooterWidth, altFooterHeight)

						' add html to the footer
						Dim altFooterHtml As New HtmlToPdfElement(0, 0, 0, pdfConverter.PdfHeaderOptions.HeaderHeight, Path.Combine(Application.StartupPath, "..\..\HeaderFooter\HeaderAndFooterHtml2.htm"), 1024, 0)
						altFooterHtml.FitHeight = True
						altFooterTemplate.AddElement(altFooterHtml)

						For pageIndex As Integer = 1 To pdfDocument.Pages.Count - 1 Step 2
							Dim pdfPage As PdfPage = pdfDocument.Pages(pageIndex)

                            pdfPage.Header = altHeaderTemplate
                            pdfPage.Footer = altFooterTemplate
						Next pageIndex
					End If

					' save the PDF document to a file on disk
					Try
						pdfDocument.Save(outFilePath)
					Finally
						' close the Document to realease all the resources
						pdfDocument.Close()
					End Try
				End If

				Dim dr As DialogResult = MessageBox.Show("Open the rendered file in an external viewer?", "Open Rendered File", MessageBoxButtons.YesNo)
				If dr = DialogResult.Yes Then
					Process.Start(outFilePath)
				End If
			Catch ex As Exception
				MessageBox.Show(ex.Message)
				Return
			Finally
				Me.Cursor = Cursors.Arrow
			End Try
		End Sub
	End Class
End Namespace

File: packages/EvoHtmlToPdf-v4.12/Samples/NET_2.0/CS/WinForms/EvoHtmlToPdfDemo/HtmlInHeaderAndFooter.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.IO;

using EvoPdf;

namespace EvoHtmlToPdfDemo
{
    public partial class HtmlInHeaderAndFooter : Form
    {
        public HtmlInHeaderAndFooter()
        {
            InitializeComponent();
        }

        private void btnConvert_Click(object sender, EventArgs e)
        {
            this.Cursor = Cursors.WaitCursor;

            try
            {
                PdfConverter pdfConverter = new PdfConverter();

                // set the license key
                pdfConverter.LicenseKey = "B4mYiJubiJiInIaYiJuZhpmahpGRkZE=";

                // show header and footer in the rendered PDF
                pdfConverter.PdfDocumentOptions.ShowHeader = true;
                pdfConverter.PdfDocumentOptions.ShowFooter = true;

                // set the header height in points
                pdfConverter.PdfHeaderOptions.HeaderHeight = 60;

                // set the header HTML area
                HtmlToPdfElement headerHtml = new HtmlToPdfElement(0, 0, 0, pdfConverter.PdfHeaderOptions.HeaderHeight,
                    Path.Combine(Application.StartupPath, @"..\..\HeaderFooter\HeaderAndFooterHtml.htm"), 1024, 0);
                headerHtml.FitHeight = true;
                pdfConverter.PdfHeaderOptions.AddElement(headerHtml);

                // set the footer height in points
                pdfConverter.PdfFooterOptions.FooterHeight = 60;
                //write the page number
                TextElement footerTextElement = new TextElement(0, 30, "This is page &p; of &P;  ",
                    new Font(new FontFamily("Times New Roman"), 10, GraphicsUnit.Point));
                footerTextElement.TextAlign = HorizontalTextAlign.Right;
                pdfConverter.PdfFooterOptions.AddElement(footerTextElement);

                // set the footer HTML area
                HtmlToPdfElement footerHtml = new HtmlToPdfElement(0, 0, 0, pdfConverter.PdfFooterOptions.FooterHeight,
                    Path.Combine(Application.StartupPath, @"..\..\HeaderFooter\HeaderAndFooterHtml.htm"), 1024, 0);
                footerHtml.FitHeight = true;
                pdfConverter.PdfFooterOptions.AddElement(footerHtml);

                // save the PDF bytes in a file on disk
                string outFilePath = Path.Combine(Application.StartupPath, "HtmlInHeaderAndFooter.pdf");

                if (!cbAlternateHeaderAndFooter.Checked)
                {
                    // the header content is the same on all the PDF pages
                    // the footer content is the same on all the PDF pages
                    pdfConverter.SavePdfFromUrlToFile(textBoxURL1.Text, outFilePath);
                }
                else
                {
                    // set an alternate header and footer on the even pages

                    // call the converter and get a Document object from URL
                    Document pdfDocument = pdfConverter.GetPdfDocumentObjectFromUrl(textBoxURL1.Text);

                    if (pdfDocument.Pages.Count >= 2)
                    {
                        // get the alternate header and footer width and height
                        // the width is given by the PDF page width
                        float altHeaderFooterWidth = pdfDocument.Pages[0].ClientRectangle.Width;
                        // the height is the same with the document header height from the PdfConverter object
                        float altHeaderHeight = pdfConverter.PdfHeaderOptions.HeaderHeight;
                        float altFooterHeight = pdfConverter.PdfFooterOptions.FooterHeight;

                        // create the alternate header template
                        Template altHeaderTemplate = pdfDocument.Templates.AddNewTemplate(altHeaderFooterWidth, altHeaderHeight);

                        // add html to the header
                        HtmlToPdfElement altHeaderHtml = new HtmlToPdfElement(0, 0, 0, pdfConverter.PdfHeaderOptions.HeaderHeight,
                            Path.Combine(Application.StartupPath, @"..\..\HeaderFooter\HeaderAndFooterHtml2.htm"), 1024, 0);
                        altHeaderHtml.FitHeight = true;
                        altHeaderTemplate.AddElement(altHeaderHtml);

                        // add a horizontal line to the bottom of the header
                        LineElement headerLine = new LineElement(0, altHeaderHeight, altHeaderFooterWidth, altHeaderHeight);
                        altHeaderTemplate.AddElement(headerLine);

                        // add page numbering to the left of the header
                        PdfFont pageNumberFont = pdfDocument.Fonts.Add(new Font(new FontFamily("Times New Roman"), 10, GraphicsUnit.Point));
                        TextElement pageNumbering = new TextElement(10, 10, "Page &p; of &P;", pageNumberFont, Color.Blue);

                        altHeaderTemplate.AddElement(pageNumbering);

                        // create the alternate footer template
                        Template altFooterTemplate = pdfDocument.Templates.AddNewTemplate(altHeaderFooterWidth, altFooterHeight);

                        // add html to the footer
                        HtmlToPdfElement altFooterHtml = new HtmlToPdfElement(0, 0, 0, pdfConverter.PdfHeaderOptions.HeaderHeight,
                            Path.Combine(Application.StartupPath, @"..\..\HeaderFooter\HeaderAndFooterHtml2.htm"), 1024, 0);
                        altFooterHtml.FitHeight = true;
                        altFooterTemplate.AddElement(altFooterHtml);

                        for (int pageIndex = 1; pageIndex < pdfDocument.Pages.Count; pageIndex += 2)
                        {
                            PdfPage pdfPage = pdfDocument.Pages[pageIndex];

                            pdfPage.Header = altHeaderTemplate;
                            pdfPage.Footer = altFooterTemplate;
                        }
                    }

                    // save the PDF document to a file on disk
                    try
                    {
                        pdfDocument.Save(outFilePath);
                    }
                    finally
                    {
                        // close the Document to realease all the resources
                        pdfDocument.Close();
                    }
                }

                DialogResult dr = MessageBox.Show("Open the rendered file in an external viewer?", "Open Rendered File", MessageBoxButtons.YesNo);
                if (dr == DialogResult.Yes)
                {
                    System.Diagnostics.Process.Start(outFilePath);
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                return;
            }
            finally
            {
                this.Cursor = Cursors.Arrow;
            }
        }
    }
}